internal/components/datasource/controller.go

```diff
@@ -24,14 +26,18 @@ type Store interface {
 	Delete(ctx context.Context, uid string) error
 }
 
-// We could also accept Bridge instead
-func ProvideDatasourceController(mgr ctrl.Manager, cli rest.Interface, stor Store) (*DatasourceReconciler, error) {
+// TODO: Looks like this should be the other way around,
+// Otherwise the reconciler will never be registered,
+// since there are no components that depend on DatasourceReconciler
+// I think we need some kind of reconciler registry (maybe just as part of the k8s bridge),
+// similar to background services registry.
```

Comment:
@papagian and i came to a similar conclusion this morning, though what we arrived at there was just extending `schema.ObjectSchema` to also include the reconciler as a quick fix. That seems like a coupling we'll probably want to break sooner rather than later, but...i mean ideally i'd love it if we could do this registration generically